src/report/service/ErrorReportingService.ts

```diff
@@ -26,17 +31,78 @@ export default class ErrorReportingService {
     );
   }
 
-  public queueError(errorReport: ErrorReport): void {
-    const channelApeOrderUrl = `https://${this.channelApeWebAppDomainName}/orders/${errorReport.channelApeOrderId}`;
-    const message = {
-      module: errorReport.module,
-      channelOrderId: errorReport.channelOrderId,
-      poNumber: errorReport.poNumber,
-      channelApeOrder: errorReport.channelApeOrderId ? channelApeOrderUrl : undefined,
-      message: errorReport.message
-    };
-    this.sqsMessageService.sendMessage(message, uuid.v4())
-      .catch(err => this.logger.error('Failed to queue the following error: ' +
-        `${JSON.stringify(message)} because: ${err}`));
+  public queueError(errorReport: ErrorReport): Promise<{ }> {
+    return new Promise((resolve, reject) => {
+      if (this.isErrorReportMissingData(errorReport)) {
+        this.getMissingData(errorReport)
+          .then(errorReport => this.sendMessageToSqs(errorReport))
+          .then(reportSent => resolve(reportSent))
+          .catch(err => reject(err));
+      } else {
+        this.sendMessageToSqs(errorReport)
+          .then(reportSent => resolve(reportSent))
+          .catch(err => reject(err));
+      }
+    });
+  }
+
+  private isErrorReportMissingData(errorReport: ErrorReport): boolean {
+    if (this.channelApeClient === undefined || errorReport.module === ErrorReportModule.INVENTORY) {
+      return false;
+    }
+    if (!errorReport.channelApeOrderId || !errorReport.channelOrderId || !errorReport.poNumber) {
+      return true;
+    }
+    return false;
+  }
+
+  private getMissingData(errorReport: ErrorReport): Promise<ErrorReport> {
+    return new Promise((resolve) => {
+      this.getOrderFromErrorReport(errorReport)
+        .then((order) => {
+          errorReport.channelApeOrderId = order.id;
+          errorReport.channelOrderId = order.channelOrderId;
+          errorReport.poNumber = AdditionalFieldService.getValue(order.additionalFields, 'name');
+          resolve(errorReport);
+        })
+        .catch(() => resolve(errorReport));
+    });
+  }
+
+  private getOrderFromErrorReport(errorReport: ErrorReport): Promise<Order> {
+    return new Promise((resolve, reject) => {
+      if (this.channelApeClient === undefined) {
+        return reject();
+      }
+      if (errorReport.channelApeOrderId) {
+        this.channelApeClient.orders()
+          .get(errorReport.channelApeOrderId)
+          .then(order => resolve(order))
+          .catch(err => reject(err));
+      } else if (errorReport.channelOrderId && this.businessId) {
+        this.channelApeClient.orders()
+          .get({ channelOrderId: errorReport.channelOrderId, businessId: this.businessId })
+          .then(orders => resolve(orders[0]))
+          .catch();
```

Comment:
Why aren't you rejecting the promise here? Please reject like you did on line 81